### PR TITLE
Fix widget state API

### PR DESCRIPTION
**Pull request by lasanthaS**
## Purpose
This PR fixes widget state API which tried to JSON parse a URI Encoded string.

### CODE REVIEW

**Comment by lasanthaS** [comment]
Once the PR is merged, the base widget needs to be published to NPM and update the version in the sample widget's package.json file.

File: base-widget/src/Widget.jsx [hunk 1]
```diff
@@ -66,8 +66,9 @@ export default class Widget extends Component {
      */
     static _getStateObject() {
         // De-serialize the object in suitable format
-        return (window.location.hash === '' || window.location.hash === '#') ?
-            {} : JSON.parse(window.location.hash.substr(1));
+        let hash = decodeURIComponent(window.location.hash);
```
Comment (this) [inline base-widget/src/Widget.jsx]: Lets use `const` here

File: base-widget/src/Widget.jsx [hunk 2]
```diff
@@ -101,14 +102,36 @@ export default class Widget extends Component {
...
+    }
+
+    /**
+     * Set global widget state.
+     *
+     * @param {string} key Key
+     * @param {{}} state State
+     */
+    setGlobalState(key, state) {
+        let states = Widget._getStateObject();
```
Comment (this) [inline base-widget/src/Widget.jsx]: Lets use `const` here as well.

File: base-widget/src/Widget.jsx [hunk 3]
```diff
@@ -101,14 +102,36 @@ export default class Widget extends Component {
...
 
+    /**
+     * Get global widget state.
+     *
+     * @param {string} key Key
+     * @returns {{}}
+     */
+    getGlobalState(key) {
+        const states = Widget._getStateObject();
+        return Object.prototype.hasOwnProperty.call(states, key) ? states[key] : { };
```
Comment (this) [inline base-widget/src/Widget.jsx]: Can't we simple `return states[key] || {}`

File: base-widget/src/Widget.jsx (hunk 2 above)
Comment (lasanthaS) [reply base-widget/src/Widget.jsx]: +1

File: base-widget/src/Widget.jsx (hunk 3 above)
Comment (lasanthaS) [reply base-widget/src/Widget.jsx]: +1

**Comment by this** [comment]
Published 1.4.0 with #1035

**Pull request merged**